proof/refine/ARM/Invariants_H.thy

```diff
@@ -50,9 +50,27 @@ definition
 definition
   "ko_wp_at' P p s \<equiv>
    \<exists>ko. ksPSpace s p = Some ko \<and> is_aligned p (objBitsKO ko) \<and> P ko \<and>
-        ps_clear p (objBitsKO ko) s"
-
-
+        ps_clear p (objBitsKO ko) s \<and> objBitsKO ko < word_bits"
+
+lemma valid_sz_simps:
+"objBitsKO ko < word_bits =
+ (case ko of
+    (KOEndpoint _) \<Rightarrow> True
+  | (KONotification _) \<Rightarrow>    True
+  | (KOCTE _) \<Rightarrow>    True
+  | (KOTCB _) \<Rightarrow>    True
+  | (KOUserData) \<Rightarrow>    True
+  | (KOUserDataDevice) \<Rightarrow>    True
+  | (KOKernelData) \<Rightarrow>    True
+  | (KOArch _) \<Rightarrow>    True
+  | (KOSchedContext sc) \<Rightarrow> scBitsFromRefillLength sc < word_bits
+  | (KOReply _) \<Rightarrow>    True
+  )"
```

Comment:
You can use a default pattern in case, e.g.:

```suggestion
lemma valid_sz_simps:
"objBitsKO ko < word_bits =
 (case ko of
  | KOSchedContext sc \<Rightarrow> scBitsFromRefillLength sc < word_bits
  | _ \<Rightarrow>    True
  )"
```